refactor(Order): change foreign key relationship

dexterklui opened the pull request:
Now an order is related to a portfolio instead of a user entity. Because
settlement should be made with a specific portfolio that is holding
actual assets, rather a user.

dexterklui committed:
refactor(Order): change foreign key relationship

Now an order is related to a portfolio instead of a user entity. Because
settlement should be made with a specific portfolio that is holding
actual assets, rather a user.

essteer commented on src/main/java/com/fdmgroup/forex/repos/OrderRepo.java:
```diff
@@ -13,8 +14,12 @@
 @Repository
 public interface OrderRepo extends JpaRepository<Order, UUID> {
 
+	@Query("SELECT o FROM Order o WHERE o.portfolio.user.id = :id")
```
Is this annotation a requirement for the operation to work? I was under the impression the JPA repository interface would handle the query in the existing format

dexterklui replied:
You're right Elliott
I think perhaps we can just rename the method to something like `findByPortfolio_User_Id` then let JPA handles the actual query.

essteer replied:
Yes if the objective is to find orders that belong to a given user ID, your line should work:

`findByPortfolio_User_Id`

I have only done a second-order query before (e.g. `findByPortfolio_User`) but in theory it should work this way

dexterklui replied:
Resolved by eebb507

dexterklui committed:
refactor(OrderRepo): use JPA naming convention

Change the method name to use JPA convenntion, to rid of need for @Query
annotation
